components/TravelPlanDetail/index.js

```diff
@@ -1,8 +1,32 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 
-const TravelPlanDetail = ({ plan }) => {
-  const sortedActivities = plan.activities.sort((a, b) => new Date(a.startDate) - new Date(b.startDate));
+const TravelPlanDetail = ({ id }) => {
+  // Create a state variable to store the fetched travel plan
+  const [plan, setPlan] = useState(null);
 
+  useEffect(() => {
+    // Define an effect to fetch the travel plan when the component mounts and every time the 'id' prop changes
+    const fetchPlan = async () => {
+      // Make a fetch request to the API endpoint that retrieves the travel plan based on the provided 'id'
+      const res = await fetch(`/api/plans/${id}`);
+      const data = await res.json();
+      setPlan(data); // Update the state with the fetched travel plan
+    };
+
+    fetchPlan(); // Call the fetchPlan function
+  }, [id]); // The effect depends on the 'id' prop
+
+  // If the travel plan hasn't been fetched yet, render a loading message
+  if (!plan) {
+    return <div>Loading...</div>;
+  }
+
+  // Sort the activities by their start dates
+  const sortedActivities = plan.activities.sort(
```

Comment:
```suggestion
  const sortedActivities = [...plan.activities].sort(
```

Reply:
You should create a copy of `plan.activities` before sorting it. This ensures that the original `plan.activities` array is not mutated.